frame/support/procedural/src/lib.rs

```diff
@@ -583,3 +584,646 @@ pub fn storage_alias(_: TokenStream, input: TokenStream) -> TokenStream {
 		.unwrap_or_else(|r| r.into_compile_error())
 		.into()
 }
+
+/// Used internally to decorate pallet attribute macro stubs when they are erroneously used
+/// outside of a pallet module
+fn pallet_macro_stub() -> TokenStream {
+	quote!(compile_error!(
+		"This attribute can only be used from within a pallet module marked with `#[frame_support::pallet]`"
+	))
+	.into()
+}
+
+/// The mandatory attribute `#[pallet::config]` defines the generics of the pallet.
+///
+/// Item must be defined as:
+///
+/// ```ignore
+/// #[pallet::config]
+/// pub trait Config: frame_system::Config + $optionally_some_other_supertraits
+/// $optional_where_clause
+/// {
+/// ...
+/// }
+/// ```
+///
+/// I.e. a regular trait definition named `Config`, with the supertrait
+/// `frame_system::pallet::Config`, and optionally other supertraits and a where clause.
```

Comment:
Mentioning "supertraits" here without really explaining why one would use them is not quite helpful. We should really state here that supertraits are often `Config` traits from other pallets, and they are used to tightly couple the current pallet with the other. Pallets that are tightly coupled can have their pallet methods accessed easily via the `super_pallet::Pallet::<T>` syntax.

Reply:
resolved with link to tight coupling article